More consistent maps

guycalledfrank opened the pull request:
- Reduces the amount of shader chunks needed for all maps;
- Supports multiplying texture * color * vertexColor (needed for GLTF)
- New property for each map: VertexColorChannel (e.g. diffuseVertexColorChannel)

willeastcott commented on src/graphics/program-lib/standard.js:
```diff
@@ -283,9 +422,10 @@ pc.programlib.standard = {
             mname = p + "Map";
             if (options[mname + "VertexColor"]) {
-                cname = mname + "Channel";
+                cname = mname + "VertChannel";
```
I think we should have:

diffuse
diffuseMap
diffuseMapChannel
diffuseVertexColor
diffuseVertexColorChannel

guycalledfrank commented:
Renamed (for all maps):
diffuseMapVertexColor -> diffuseVertexColor
diffuseMapVertChannel -> diffuseVertexColorChannel
diffuseMapTint -> diffuseTint

Also updated docs

daredevildave commented:
This is a breaking change right?

guycalledfrank commented:
With all the renaming it is. Before the renaming it wasn't.
... can add get/setters for old property names

daredevildave commented on src/graphics/program-lib/chunks/aoMap.frag:
```diff
@@ -0,0 +1,18 @@
+#ifdef MAPTEXTURE
```
This define applies to all maps? How come we don't need one for each map type?

guycalledfrank replied:
No. Shader generator #undefs previous defines before every map: https://github.com/playcanvas/engine/pull/1080/files#diff-0df44c31b04313738bd9cf0de37958bbR173

daredevildave commented on src/graphics/program-lib/standard.js:
```diff
@@ -1,4 +1,98 @@
 pc.programlib.standard = {
+
+    _oldChunkToNew: {
+        aoTexPS : {n:"aoMapPS", f:_oldChunkTex},
```
If we doing all this work to support old chunk names, can we get rid of "PS" as GLSL doesn't use pixel shaders. And also doesn't fit our coding standards for acronyms (should be `aoTexFs`?)

guycalledfrank replied:
Next time, in another PR ;)

daredevildave replied:
Then we're going to break any shader chunk code twice?

guycalledfrank replied:
This PR is made carefully to NOT break any shader chunk code.

guycalledfrank replied:
But doing setters for all chunks we have for the sake of changing names isn't useful

daredevildave replied:
So it doesn't break shader chunk code, but does break javascript?

guycalledfrank replied:
diffuseMapTint, diffuseMapVertexColor -> diffuseTint, diffuseVertexColor.

This is breaking, because you could use old names in procedural material creation code. But I can define old names as get/set too.

guycalledfrank commented:
OK, it's not longer breaking.

willeastcott commented:
OK, all looks good to me. Merge?

guycalledfrank commented:
Why not?
I tested various stuff with it, but you might try testing some other stuff too + make sure it does work with GLTF the way it should.

willeastcott commented:
Checked against the glTF sample model test suite. Works fine! Merging...
